src/snowflake/connector/cursor.py

```diff
@@ -117,6 +117,19 @@ class ResultMetadata(NamedTuple):
     scale: int
     is_nullable: bool
 
+    @classmethod
+    def from_column(cls, col):
```

Comment:
Type-hint for `col` please. I know that `json.loads` can give you a lot of things, but I think we can well-define at least the key part of it.

Also please add a one-liner docstring describing what this is for.